src/viur/core/secret.py

```diff
@@ -0,0 +1,27 @@
+import logging
+from google.cloud import secretmanager
+from viur.core.config import conf
+
+"""
+This module provides utiliy functions for accessing values stored in the Google Cloud Secret Manager.
+"""
+
+# Global secret manager client instance
+__client = secretmanager.SecretManagerServiceClient()
+
+
+def get(secret: str, version: str = "latest") -> str:
+    """
+    Retrieves a secret stored in Google Cloud Secret Manager for use within the application.
+
+    Add a secret online under https://console.cloud.google.com/security/secret-manager.
+    Service accounts requires the role "Secret Manager Secret Accessor" in IAM.
+    """
+    name = f"""projects/{conf["viur.instance.project_id"]}/secrets/{secret}/versions/{version}""".replace(".", "_")
+
+    try:
+        return __client.access_secret_version(request={"name": name}).payload.data.decode()
+    except Exception as e:
+        logging.exception(e)
+
+    return None
```

Comment:
I think silent error is a stupid idea. Exceptions are there to send signals and not to be ignored. If I explicitly call a function, I also want to get feedback in case of an error.

Imagine you use a zip function (which creates a zip archive), which has the parameter `password`. As argument you pass the return value of the `secret.get` method.
If your secret is not set or the permissions are set incorrectly the zip archive might be created without password protection (because `None` might be the input for "no password protection"). You will only see later in the logs that your software has already distributed open zips instead of aborting the process with an exception.

```suggestion
    return __client.access_secret_version(request={"name": name}).payload.data.decode()
```